src/SSHLibrary/abstractclient.py

```diff
@@ -202,7 +202,9 @@ def login(self, username=None, password=None, allow_agent=False, look_for_keys=F
         """
         keep_alive_interval = int(TimeEntry(keep_alive_interval).value)
         username = self._encode(username)
-        if not password and not allow_agent:
+        if not password:
+            password = None
```

Comment:
I am wondering if this is the correct solution.
Is `password = ""` and `password == None` the same for paramiko and trilead?

The reason why `Login user ${EMPTY}` doesn't work anymore is that 355ff5ca7d changed `if password == None" to `if not password` in [pythonclient.py](https://github.com/robotframework/SSHLibrary/commit/355ff5ca7db2e4e52d611b42e3719166d79c31e6#diff-081bceef8d5f9d82a4301e8dabda221373afb0a22a59f560e7b20e72a894fed6). This changes semantics, because now `password == ""` and `password == None` use this code path. Previously `password == ""` used the else-path and that worked for empty passwords